routes/index.js

```diff
@@ -118,22 +117,72 @@ function formatUserData (res) {
       flist[i] = fname + ' ' + lname
     }
   }
+  userData.id = friends[0].id
   userData.firstname = friends[0].firstname.charAt(0).toUpperCase() + friends[0].firstname.slice(1)
   userData.lastname = friends[0].lastname.charAt(0).toUpperCase() + friends[0].lastname.slice(1)
   userData.friendslist = flist
   return userData
 }
 
 // Route for adding friends
-router.post('/api/friendships', function (req, res) {
-  // const { firstname, lastname, email } = req.body
-
-  // check if duplicate
-
-  // add user to users table
-
-  // add friendship to friendship table (friender id, id of added friend)
+router.post('/api/friendships', async function (req, res) {
+  const { userid, friendfname, friendlname, friendemail } = req.body
+  const secret = uuidv4()
 
+  // Checks to see if friendee is already a user
+  const result = await emailAlreadyRegistered(friendemail)
+  if (!result) { // Friend is not a user, insert user into query and return user id
+    const insertUser = {
+      text: 'INSERT INTO users(firstname, lastname, email, secret) VALUES ($1, $2, $3, $4) RETURNING id',
+      values: [friendfname.toLowerCase(), friendlname.toLowerCase(), friendemail.toLowerCase(), secret]
+    }
+    db.query(insertUser)
+      .then(result => {
+        addFriend(userid, result.rows[0].id)
+        res.status(200).send()
+      })
+      .catch(err => {
+        res.status(500).json(err)
+      })
+  } else { // friend is already a user, check if friendship alreayd exists. If not, add friendship
+    const checkExistingFriendship = {
+      text: 'SELECT user1, u1.firstname, u1.email, user2, u2.firstname, u2.email FROM friendships ' +
+      'LEFT JOIN users as u1 ON  friendships.user1 = u1.id ' +
+      'LEFT JOIN users as u2 ON friendships.user2 = u2.id WHERE user1 = $1 AND u2.email = $2',
+      values: [userid, friendemail]
+    }
+    db.query(checkExistingFriendship)
+      .then(result => {
+        if (result.rows.length !== 0) { // If length of rows  != 0, that means there is an existing friendship between 2 users
+          res.status(409).send() // I still cant read message after a 409 error should I maybe use a
+        } else {
+          const getFriendId = { // gets friend user id from email
+            text: 'SELECT id FROM users WHERE users.email = $1',
+            values: [friendemail]
+          }
+          db.query(getFriendId)
+            .then(result => {
+              addFriend(userid, result.rows[0].id)
+              res.status(200).send()
+            })
+            .catch(err => {
+              res.status(500).json(err)
+            })
+        }
+      })
+      .catch(err => {
+        res.status(500).json(err)
+      })
+  }
 })
 
+// Helper function for /api/friendships to add friends
+function addFriend (user1, user2) {
+  const addFriendship = {
+    text: 'INSERT INTO friendships(user1, user2) VALUES ($1, $2) ',
+    values: [user1, user2]
+  }
+  db.query(addFriendship)
```

Comment:
[nit] Don't forget to return at the end, that way you can use `await` or `.then` on the promise.